p11/slot.go

```diff
@@ -28,6 +28,12 @@ func (s Slot) OpenWriteSession() (Session, error) {
 	return s.openSession(pkcs11.CKF_RW_SESSION)
 }
 
+// OpenExclusiveWriteSession opens an exclusive read-write session with the
+// token in this slot.
+func (s Slot) OpenExclusiveWriteSession() (Session, error) {
+	return s.openSession(pkcs11.CKF_RW_SESSION | pkcs11.CKF_EXCLUSIVE_SESSION)
```

Comment:
We should probably add a opensession that takes flags for folks that need it, instead of adding new functions for every combination.

Reply:
I thought in doing that at first, but there's already a `OpenSession` method that takes no arguments and I couldn't come up with a good name for the new function. What do you think about `OpenSessionWithFlags` or `OpenSessionUsingFlags`?